File: packages/python/examples/pytest/file_upload_test.py
from os import getenv
from tempfile import NamedTemporaryFile

from pytest import fixture, mark

from alumnium import Model, Provider

driver_type = getenv("ALUMNIUM_DRIVER", "selenium")


@fixture
def file():
    with NamedTemporaryFile(mode="w+", suffix=".txt") as file:
        yield file


@fixture
def file2():
    with NamedTemporaryFile(mode="w+", suffix=".txt") as file:
        yield file


@mark.xfail("appium" in driver_type, reason="File upload is not implemented in Appium yet")
def test_file_upload(al, file, navigate):
    navigate("https://the-internet.herokuapp.com/upload")
    al.do(f"upload '{file.name}'")
    al.do("click on 'Upload' button")
    assert al.get("heading") == "File Uploaded!"


@mark.xfail("appium" in driver_type, reason="File upload is not implemented in Appium yet")
@mark.xfail(Model.current.provider == Provider.AWS_META, reason="Prefers to click on the upload button manually")
def test_multiple_file_upload(al, file, file2, navigate):
    navigate("multiple_file_upload.html")
    al.do(f"upload files '{file.name}', '{file2.name}'")
    al.do("click 'Upload Files' button")
    assert al.get("success message") == "✓ Upload Successful!"
    assert al.get("uploaded files names") == [
        file.name.split("/")[-1],
        file2.name.split("/")[-1],
    ]


@mark.xfail("appium" in driver_type, reason="File upload is not implemented in Appium yet")
@mark.xfail(driver_type == "selenium", reason="Hidden file upload inputs are not supported in Selenium")
@mark.xfail(Model.current.provider == Provider.AWS_META, reason="Prefers to click on the upload button manually")
def test_hidden_file_upload(al, file, navigate):
    navigate("hidden_file_upload.html")
    al.do(f"upload '{file.name}' to 'Choose Files' button")
    al.do("click 'Upload Files' button")
    assert al.get("success message") == "Files Uploaded Successfully!"

File: packages/typescript/examples/mocha/fileUpload.test.ts
import assert from "assert";
import { unlinkSync, writeFileSync } from "fs";
import { tmpdir } from "os";
import { join } from "path";
import "./globals.js";
import { navigate } from "./helpers.js";

const driverType = process.env.ALUMNIUM_DRIVER || "selenium";

describe("File Upload", () => {
  let testFile1: string;
  let testFile2: string;

  beforeEach(function () {
    const timestamp = Date.now();
    testFile1 = join(tmpdir(), `test-upload-${timestamp}-1.txt`);
    testFile2 = join(tmpdir(), `test-upload-${timestamp}-2.txt`);
    writeFileSync(testFile1, "Test content 1");
    writeFileSync(testFile2, "Test content 2");
  });

  afterEach(function () {
    unlinkSync(testFile1);
    unlinkSync(testFile2);
  });

  it("should upload a single file", async function () {
    if (driverType === "appium") {
      this.skip();
    }

    await navigate(driver, "https://the-internet.herokuapp.com/upload");
    await al.do(`upload '${testFile1}'`);
    await al.do("click on 'Upload' button");
    const heading = await al.get("heading");
    assert.strictEqual(heading, "File Uploaded!");
  });

  it("should upload multiple files", async function () {
    if (driverType === "appium") {
      this.skip();
    }

    await navigate(driver, "multiple_file_upload.html");
    await al.do(`upload files '${testFile1}', '${testFile2}'`);
    await al.do("click 'Upload Files' button");
    const message = await al.get("success message");
    assert.strictEqual(message, "✓ Upload Successful!");
    const uploadedFiles = await al.get("uploaded files names");
    assert.deepStrictEqual(uploadedFiles, [
      testFile1.split("/").pop(),
      testFile2.split("/").pop(),
    ]);
  });
});
